pandas/core/arrays/arrow/array.py

```diff
@@ -1084,7 +1083,6 @@ def fillna(
         copy: bool = True,
     ) -> Self:
         if not self._hasna:
-            # TODO(CoW): Not necessary anymore when CoW is the default
             return self.copy()
 
```

Comment:
Why not ´return self´?

Reply:
We have unit testing that asserts that `ExtensionArray.fillna` should return a copy even if a no-op